src/goto-synthesizer/goto_synthesizer_parse_options.cpp

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************\
+Module: Parse Options Module
+Author: Michael Tautschnig
+\*******************************************************************/
+
+/// \file
+/// Main Module
+
+#include "goto_synthesizer_parse_options.h"
+
+#include <util/config.h>
+#include <util/exit_codes.h>
+#include <util/version.h>
+
+#include <goto-programs/read_goto_binary.h>
+
+#include <iostream>
+
+/// invoke main modules
+int goto_synthesizer_parse_optionst::doit()
+{
+  if(cmdline.isset("version"))
+  {
+    std::cout << CBMC_VERSION << '\n';
+    return CPROVER_EXIT_SUCCESS;
+  }
+
+  if(cmdline.args.size() != 1 && cmdline.args.size() != 2)
+  {
+    help();
+    return CPROVER_EXIT_USAGE_ERROR;
+  }
+
+  messaget::eval_verbosity(
+    cmdline.get_value("verbosity"), messaget::M_STATISTICS, ui_message_handler);
+
+  register_languages();
+
+  get_goto_program();
+
+  // TODO
+  // Migrate synthesizer and tests from goto-instrument to goto-synthesizer
+
+  help();
+  return CPROVER_EXIT_USAGE_ERROR;
+}
+
+void goto_synthesizer_parse_optionst::get_goto_program()
+{
+  log.status() << "Reading GOTO program from '" << cmdline.args[0] << "'"
+               << messaget::eom;
+
+  config.set(cmdline);
+
+  auto result = read_goto_binary(cmdline.args[0], ui_message_handler);
+
+  if(!result.has_value())
+    throw 0;
```

Comment:
We might be doing this elsewhere, but moving forward I don't think throwing exceptions in this piece of code is the right thing to do. You probably want to add a return value to this method, return `CPROVER_EXIT_USAGE_ERROR`, and have the caller `return` on any non-zero return value.